### DISCUSSION TITLE
Set Fixed Cost of Vehicles

### DISCUSSION THREAD

**Discussion by Debom13** [google/or-tools#2580] [Routing (and legacy CP) questions] [answered]
Hi.... I want to assign fixed cost to the vehicles. But event after using "SetFixedCostOfVehicle()" function, it doesn't seems to be working.
```python
    for vehicle_id, vehicle_cost in enumerate(data['vehicle_capacities']):
      if vehicle_id > 3:
        print(vehicle_id)
        routing.SetFixedCostOfVehicle(20, vehicle_id)
      elif vehicle_id > 3:
        routing.SetFixedCostOfVehicle(40, vehicle_id)
```
here is the snippet of the code I used. Kindly let me know how to solve the problem?
Do I need to change `routing.SetArcCostEvaluatorOfAllVehicles(transit_callback_index)` ?

**Comment by Mizux** [comment, marked as answer]
your `if` and `elif` use the same condition this is weird to me...

```py
if cond 1:
  ... #cond1 valid
elif cond 2:
  ... #not cond1 and cond2
else
  .... # default not cond1 nor cond2
```

**Reply by Debom13** [reply]
Sorry, I have pasted the wrong code by mistake.

     for vehicle_id, vehicle_cost in enumerate(data['vehicle_capacities']):
       if vehicle_id <= 3:
         routing.SetFixedCostOfVehicle(20, vehicle_id)
       elif vehicle_id > 3:
         routing.SetFixedCostOfVehicle(40, vehicle_id)

This is the one that I have used

**Reply by Debom13** [reply]
Actually I wanted to set the vehicles in a particular order, that is why using this method. Is there any better method to do so?
